rotkehlchen/accounting/cost_basis.py

```diff
@@ -423,6 +465,115 @@ def calculate_spend_cost_basis(
             is_complete=is_complete,
         )
 
+    def calculate_spend_cost_basis_wac(
+            self,
+            spending_amount: FVal,
+            spending_asset: Asset,
+            timestamp: Timestamp,
+    ) -> CostBasisInfo:
+        """
+        When spending `spending_amount` of `spending_asset` at `timestamp` this function
+        calculates using the wac method the corresponding buy/s from
+        which to do profit calculation. Does NOT apply the "free after given time period"
+        rule which applies for some jurisdictions such as 1 year for Germany.
+
+        Returns the information in a CostBasisInfo object if enough acquisitions have
+        been found.
+        """
+        taxfree_bought_cost = ZERO
+        taxable_bought_cost = ZERO
+        taxable_amount = ZERO
+        is_complete = True
+        asset_events = self.get_events(spending_asset)
+
+        # for WAC, we need to rebuild the position up until timestamp. This is expensive.
+
+        # init a list of buy and sell we will use to build the pos
+        all_events = []
+
+        # add all buy that happened before timestamp
+        for acquisition_event in asset_events.acquisitions:
+            if acquisition_event.timestamp <= timestamp:
+                all_events.append(
+                    AssetEvent(
+                        acquisition_event.timestamp,
+                        acquisition_event.location,
+                        acquisition_event.description,
+                        acquisition_event.amount,
+                        FVal(0),
+                        acquisition_event.rate,
+                        acquisition_event.fee_rate,
+                        FVal(0),
+                        FVal(0),
+                    ),
+                )
+
+        # add all sells that happened before timestamp
+        for spend_event in asset_events.spends:
+            if spend_event.timestamp <= timestamp:
+                all_events.append(
+                    AssetEvent(
+                        spend_event.timestamp,
+                        spend_event.location,
+                        '',
```

Comment:
Spend events don't have a description? We should add one.

Reply:
yes, what should we put there?